src/components/Grid/methods.js

```diff
@@ -1041,97 +1060,93 @@ export default {
+    // const snappableAngles = [-180, -90, 0, 90, 180];
+    // // angle between projection and mouse (degrees)
+    // const thetaDeg = Math.atan2(point.y - nearestEdge.projection.y, point.x - nearestEdge.projection.x)
+    // * (180 / Math.PI);
+
+    // // if the original projection is within the snap tolerance of one of the snapping angles
+    // // adjust the projection so that it is exactly at the snap angle
+    // // snap to -180, -90, 0, 90, 180
+    // snappableAngles.some((angle) => {
+    //   // if the original projection is within the snap tolerance of one of the snapping angles
+    //   // adjust the projection so that it is exactly at the snap angle
+    //   if (Math.abs(thetaDeg - angle) < this.$store.getters['project/angleTolerance']) {
+    //     // infer a line defining the desired projection
+    //     var adjustedProjectionP1;
+    //     var adjustedProjectionP2;
+    //     if (angle === 180 || angle === 0 || angle === -180) {
+    //       adjustedProjectionP1 = { x: point.x - (2 * nearestEdge.dist), y: point.y }
+    //       adjustedProjectionP2 = { x: point.x + (2 * nearestEdge.dist), y: point.y }
+    //     } else if (angle === 90 || angle === -90) {
+    //       adjustedProjectionP1 = { x: point.x, y: point.y - (2 * nearestEdge.dist) }
+    //       adjustedProjectionP2 = { x: point.x, y: point.y + (2 * nearestEdge.dist) }
+    //     }
+    //     // adjust the projection to be the intersection of the desired projection line and the nearest edge
+    //     if (geometryHelpers.ptsAreCollinear(adjustedProjectionP1, nearestEdgeV1, adjustedProjectionP2)) {
+    //       projection = nearestEdgeV1;
+    //     } else if (geometryHelpers.ptsAreCollinear(adjustedProjectionP1, nearestEdgeV2, adjustedProjectionP2)) {
+    //       projection = nearestEdgeV2;
+    //     } else {
+    //       projection = geometryHelpers.intersectionOfLines(adjustedProjectionP1, adjustedProjectionP2, nearestEdgeV1, nearestEdgeV2);
+    //     }
+    //     return true;
+    //   }
+    //   return false;
+    // });
 
     // return data for the edge if the projection is within the snap tolerance of the point
-    if (dist < this.$store.getters['project/snapTolerance']) {
-      return {
+    if (nearestEdge.dist < this.$store.getters['project/snapTolerance']) {
+      return [{
         snappingEdge: nearestEdge,
         dist: nearestEdge.dist,
         type: 'edge',
         // projection and snapping edge vertices translated into grid coordinates (to display snapping point and highlight edges)
-        projection,
-        v1GridCoords: nearestEdgeV1,
-        v2GridCoords: nearestEdgeV2,
-      }
+        projection: nearestEdge.projection,
+        v1GridCoords: nearestEdge.v1Coords,
+        v2GridCoords: nearestEdge.V2Coords,
+        x: nearestEdge.projection.x,
+        y: nearestEdge.projection.y,
+      }];
     }
-    return null;
+    return [0]; // ????
```

Comment:
I had been thinking that this was always a `[0]`, but it looks like that's new in this PR. Let's instead return an empty list here.